src/perf_tool_backend.ml

```diff
@@ -156,13 +157,24 @@ module Recording = struct
       | None -> []
       | Some snapshot_size -> [ [%string "-m,%{Pow2_pages.num_pages snapshot_size#Int}"] ]
     in
+    let pid_opt = [ Pid.to_int pid |> Int.to_string ] in
+    let snapshot_opt =
+      if full_execution
```

Comment:
Hmm, is `full_execution` any different from `Magic_trace_or_the_application_terminates`? Could the code paths be merged? (Not asking you to do so in this PR)

Reply:
It is if the application runs longer than a snapshot buffer length.

Reply:
Ah, right.